pkg/cli/clients/clients.go

```diff
@@ -137,40 +137,74 @@ type LogStream struct {
 
 // ApplicationsManagementClient is used to interface with management features like listing resources by app, show details of a resource.
 type ApplicationsManagementClient interface {
-	ListAllResourcesByType(ctx context.Context, resourceType string) ([]generated.GenericResource, error)
-	ListAllResourcesOfTypeInApplication(ctx context.Context, applicationName string, resourceType string) ([]generated.GenericResource, error)
-	ListAllResourcesByApplication(ctx context.Context, applicationName string) ([]generated.GenericResource, error)
-	ListAllResourcesOfTypeInEnvironment(ctx context.Context, environmentName string, resourceType string) ([]generated.GenericResource, error)
-	ListAllResourcesByEnvironment(ctx context.Context, environmentName string) ([]generated.GenericResource, error)
-	ShowResource(ctx context.Context, resourceType string, resourceName string) (generated.GenericResource, error)
-	DeleteResource(ctx context.Context, resourceType string, resourceName string) (bool, error)
+	// ListResourcesOfType lists all resources of a given type in the configured scope.
+	ListResourcesOfType(ctx context.Context, resourceType string) ([]generated.GenericResource, error)
+
+	// ListResourcesOfTypeInApplication lists all resources of a given type in a given application in the configured scope.
+	ListResourcesOfTypeInApplication(ctx context.Context, applicationNameOrID string, resourceType string) ([]generated.GenericResource, error)
+
+	// ListResourcesOfTypeInEnvironment lists all resources of a given type in a given environment in the configured scope.
+	ListResourcesOfTypeInEnvironment(ctx context.Context, environmentNameOrID string, resourceType string) ([]generated.GenericResource, error)
+
+	// ListResourcesInApplication lists all resources in a given application in the configured scope.
+	ListResourcesInApplication(ctx context.Context, applicationNameOrID string) ([]generated.GenericResource, error)
+
+	// ListResourcesInEnvironment lists all resources in a given environment in the configured scope.
+	ListResourcesInEnvironment(ctx context.Context, environmentNameOrID string) ([]generated.GenericResource, error)
+
+	// GetResource retrieves a resource by its type and name (or id).
+	GetResource(ctx context.Context, resourceType string, resourceNameOrID string) (generated.GenericResource, error)
+
+	// DeleteResource deletes a resource by its type and name (or id).
+	DeleteResource(ctx context.Context, resourceType string, resourceNameOrID string) (bool, error)
```

Comment:
There are no tests to test by ID

Reply:
I'll double check this. I don't plan to add a "name test" + "id test" for every API. The difference between the name case and the ID case are handled in common code.

Reply:
I looked again and there's a good mix of tests that use names and tests that use IDs. It feels like a lot of wasted effort to test both cases for each of the 15 functions. They all call into the same code to resolve the full id. 

Please let me know if there's something specific you want to see added.